lib/normalize.js

```diff
@@ -0,0 +1,90 @@
+
+/**
+ * Module Dependencies.
+ */
+
+var indexof = require('component/indexof');
+var defaults = require('defaults');
+var map = require('component/map');
+var each = require('each');
+var is = require('is');
+
+/**
+ * Expose `normalize`
+ */
+
+module.exports = normalize;
+
+/**
+ * Toplevel properties.
+ */
+
+var toplevel = [
+  'integrations',
+  'anonymousId',
+  'properties',
+  'previousId',
+  'timestamp',
+  'category',
+  'context',
+  'groupId',
+  'userId',
+  'traits',
+  'event',
+  'name'
+];
+
+/**
+ * Normalize `msg` based on integrations `list`.
+ *
+ * @param {Object} msg
+ * @param {Array} list
+ * @return {Function}
+ */
+
+function normalize(msg, list){
+  var lower = map(list, function(s){ return s.toLowerCase(); });
+  var opts = msg.options || {};
+  var integrations = opts.integrations || {};
+  var providers = opts.providers || {};
+  var context = opts.context || {};
+  var ret = {};
+
+  // integrations.
+  each(opts, function(key, value){
+    if (!integration(key)) return;
+    integrations[key] = value;
+    delete opts[key];
+  });
+
+  // providers.
+  delete opts.providers;
+  each(providers, function(key, value){
+    if (!integration(key)) return;
+    if (is.object(integrations[key])) return;
```

Comment:
can you elaborate? kinda confused, to give you some context on what i'm trying to do here:

```js
analytics.track('event', {}, {
  Segment: true,
  providers: {
    Segment: { ... }
  }
});
```

Just to pick up the object, but if i check for null, it will override booleans, which would just be weird i think?

```js
analytics.track('event', {}, {
  Segment: true,
  providers: {
    Segment: true,
  }
});
```

i'm sure i'm missing something haha